Sources/System/MachPort.swift

```diff
@@ -81,17 +83,18 @@ public enum Mach {
     }
 
     deinit {
-      if _name == 0xFFFFFFFF /* MACH_PORT_DEAD */ {
-        precondition(RightType.self != ReceiveRight.self, "Receive rights cannot be dead names")
-        _machPrecondition(mach_port_mod_refs(mach_task_self_, _name, MACH_PORT_RIGHT_DEAD_NAME, -1))
+      if RightType.self == ReceiveRight.self {
+        precondition(_name != 0xFFFFFFFF /* MACH_PORT_DEAD */,
+                     "Receive rights cannot be dead names")
+        _machPrecondition(
+          mach_port_destruct(mach_task_self_, _name, 0, _context)
+        )
       } else {
-        if RightType.self == ReceiveRight.self {
-          _machPrecondition(mach_port_destruct(mach_task_self_, _name, -1, _context))
-        } else if RightType.self == SendRight.self {
-          _machPrecondition(mach_port_mod_refs(mach_task_self_, _name, MACH_PORT_RIGHT_SEND, -1))
-        } else if RightType.self == SendOnceRight.self {
-          _machPrecondition(mach_port_mod_refs(mach_task_self_, _name, MACH_PORT_RIGHT_SEND_ONCE, -1))
-        }
+        assert(
+          RightType.self == SendRight.self ||
+          RightType.self == SendOnceRight.self
+        )
+        _machPrecondition(mach_port_deallocate(mach_task_self_, _name))
```

Comment:
This is a good point. I went this way because it had ballooned to 6 branches; there is probably a reasonable alternative somewhere in the middle.

Reply:
A note from an out-of-band conversation: the previously-used `mach_port_mod_refs()` calls do not succeed when `_name` has become a dead name, because it is no longer a send right (or a send-once right.) It is unclear whether there's a reasonable and performant way to re-validate the type of `_name` in the deinit.

Reply:
If we find a better solution for the `deinit`, we'll update it in another PR.